lib/fileTools.js

```diff
@@ -154,6 +154,50 @@ function searchFile (dir, fileName, callback) {
   });
 }
 
+// uses the folderScanSync to DFS and grab absolute paths of all files and folders, deleting folders (deepest first) and moving files to the target
+// requires an absolute path for sourcePath and targetFolderPath.
+function flattenFolderStructure(sourcePath, targetFolderPath, callback) {
+  return Q.resolve().then(function () {
+    var output = {
+      files: [],
+      folders: []
+    }
+    folderScanSync(output, sourcePath);
+
+    // move/copy files
+    output.files.forEach(file => {
+      fs.renameSync(file, targetFolderPath +  path.parse(file).base, (err) => {
+        return callback(err);
+      });
+    });
+
+    // remove directories
+    output.folders.forEach(folder => {
+      fs.rmdirSync(folder, (err) => {
+        callback(err);
+      });
+    });
+
+    return Q.resolve();
+  }).nodeify(callback);
+}
+
+// DFS crawl into the folder structure and grab all references.
+// output is an object: { files: [], folders: [] }
+function folderScanSync(output, directory) {
+  fs.readdirSync(directory)
+    .forEach(function (fileEntry) {
+      var fileEntryPath = path.join(directory, fileEntry);
+
+      if (fs.statSync(fileEntryPath).isDirectory()) {
+        folderScanSync(output, fileEntryPath + "/");
```

Comment:
Recursion here could get ugly for large, deeply nested directories, possibly resulting in stack overflow. Have you considered non-recursive algorithms?